### DISCUSSION TITLE
Injecting the same store in different files

### DISCUSSION THREAD

**Discussion by bauerbua** [state-adapt/state-adapt#48] [Q&A] [answered]
If I want to use the same store in multiple components and I initialize the store like the following:

```typescript
export class StoreA {
  public store = adapt(['storeA', storeAInitialState], storeAdapter)
}
```

and use it in two components like

```typescript
export class Component1 {
  public storeA = inject(StoreA).store;
}
```

```typescript
export class Component2 {
  public storeA = inject(StoreA).store;
}
```

I would create two instances of the same store right? Would the best way to share the same state be, to introduce sources for the actions and pass the selector observables by public variables inside of the `StoreA`class? It seems kind of redundant to me to introduce variables for the selectors. Especially if there are a lot of them.

**Comment by mfp22** [comment, marked as answer]
Nope, it creates one store, so both components use the same state.
